SQL/New_patches/2025_02_05_change_candid_fk_to_id.sql

```diff
@@ -0,0 +1,88 @@
+-- Change FKs from CandID to CandidateID which is now candidate.ID
+ALTER TABLE CNV DROP constraint CNV_ibfk_3;
+UPDATE CNV SET CandID=(SELECT ID from candidate c WHERE c.CandID=CNV.CandID);
+ALTER TABLE CNV CHANGE CandID CandidateID int(10) unsigned;
+ALTER TABLE CNV ADD CONSTRAINT CNV_ibfk_3 FOREIGN KEY (CandidateID) REFERENCES candidate(ID);
+
+ALTER TABLE candidate_consent_rel DROP CONSTRAINT `FK_candidate_consent_rel_CandidateID`;
+UPDATE candidate_consent_rel SET CandidateID=(SELECT ID from candidate c WHERE c.CandID=candidate_consent_rel.CandidateID);
+ALTER TABLE candidate_consent_rel CHANGE CandidateID CandidateID int(10) unsigned;
+ALTER TABLE candidate_consent_rel ADD CONSTRAINT FK_candidate_consent_rel_CandidateID FOREIGN KEY (CandidateID) REFERENCES candidate(ID) ON DELETE RESTRICT ON UPDATE RESTRICT;
+
+ALTER TABLE participant_status DROP CONSTRAINT `fk_participant_status_3`;
+UPDATE participant_status SET CandID=(SELECT ID from candidate c WHERE c.CandID=participant_status.CandID);
+ALTER TABLE participant_status CHANGE CandID CandidateID int(10) unsigned;
+ALTER TABLE participant_status ADD CONSTRAINT FK_participant_status_3 FOREIGN KEY (CandidateID) REFERENCES candidate(ID) ON DELETE CASCADE ON UPDATE CASCADE;
+
+ALTER TABLE dataquery_run_results DROP CONSTRAINT `dataquery_run_results_ibfk_1`;
+UPDATE dataquery_run_results SET CandID=(SELECT ID from candidate c WHERE c.CandID=dataquery_run_results.CandID);
+ALTER TABLE dataquery_run_results CHANGE CandID CandidateID int(10) unsigned;
+ALTER TABLE dataquery_run_results ADD CONSTRAINT dataquery_run_results_ibfk_1 FOREIGN KEY (CandidateID) REFERENCES candidate(ID);
+
+ALTER TABLE issues DROP CONSTRAINT `fk_issues_3`;
+UPDATE issues SET CandID=(SELECT ID from candidate c WHERE c.CandID=issues.CandID);
+ALTER TABLE issues CHANGE CandID CandidateID int(10) unsigned;
+ALTER TABLE issues ADD CONSTRAINT fk_issues_3 FOREIGN KEY (CandidateID) REFERENCES candidate(ID);
+
+ALTER TABLE session DROP CONSTRAINT `fk_session_1`;
+UPDATE session SET CandID=(SELECT ID from candidate c WHERE c.CandID=session.CandID);
+ALTER TABLE session CHANGE CandID CandidateID int(10) unsigned;
+ALTER TABLE session ADD CONSTRAINT fk_session_1 FOREIGN KEY (CandidateID) REFERENCES candidate(ID);l
```

Comment:
Looks like there's an extra character at the end of this line.

Reply:
Fixed